lib/pretty/json/serializer.lua

```diff
@@ -29,8 +29,26 @@ local function escape_str(s)
     return s
 end
 
+local function sorted_pairs(obj)
+    local sorted_keys = {}
+    for key in pairs(obj) do
+        table.insert(sorted_keys, key)
+    end
+
+    table.sort(sorted_keys)
+
+    local i = 0
+    return function()
+        if i < #sorted_keys then
+            i = i + 1
+            return sorted_keys[i], obj[sorted_keys[i]]
+        end
+    end
```

Comment:
You can use a coroutine for this. Something like:

```lua
return coroutine.wrap(function()
  for k, v in ipairs(sorted_keys) do
    coroutine.yield(k, v)
  end
end)
```

Reply:
This is slick